File: services/user-service/src/main/java/com/visor/school/userservice/repository/UserPermissionRepository.java
package com.visor.school.userservice.repository;

import java.util.List;
import java.util.UUID;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.visor.school.userservice.model.User;
import com.visor.school.userservice.model.UserPermission;

@Repository
public interface UserPermissionRepository extends JpaRepository<UserPermission, UUID> {
    List<UserPermission> findByUser(User user);
    List<UserPermission> findByUserId(UUID userId);
    boolean existsByUserIdAndPermissionId(UUID userId, UUID permissionId);
    void deleteByUserIdAndPermissionId(UUID userId, UUID permissionId);
}

File: services/user-service/src/main/kotlin/com/visor/school/userservice/repository/UserPermissionRepository.kt
package com.visor.school.userservice.repository

import com.visor.school.userservice.model.User
import com.visor.school.userservice.model.UserPermission
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface UserPermissionRepository : JpaRepository<UserPermission, UUID> {
    fun findByUser(user: User): List<UserPermission>
    fun findByUserId(userId: UUID): List<UserPermission>
    fun existsByUserIdAndPermissionId(userId: UUID, permissionId: UUID): Boolean
    fun deleteByUserIdAndPermissionId(userId: UUID, permissionId: UUID)
}

